OpenFF Toolkit 0.12.0 breaks some tests

```console
=========================== short test summary info ============================
FAILED openff/qcsubmit/tests/test_datasets.py::test_dataset_linear_dihedral_validator - ValueError: atomic number must be positive, given 0.
FAILED openff/qcsubmit/tests/test_datasets.py::test_torsiondrive_unqiue_settings - ValueError: atomic number must be positive, given 0.
FAILED openff/qcsubmit/tests/test_factories.py::test_torsiondrive_torsion_string - ValueError: atomic number must be positive, given 0.
FAILED openff/qcsubmit/tests/test_workflow_components.py::test_wbo_fragmentation_apply - assert 0 == 4
 +  where 0 = ComponentResult(name=WBOFragmenter, molecules=0, filtered=1).n_molecules
FAILED openff/qcsubmit/tests/test_workflow_components.py::test_pfizer_fragmentation_apply - assert 0 == 4
 +  where 0 = ComponentResult(name=PfizerFragmenter, molecules=0, filtered=1).n_molecules
====== 5 failed, 470 passed, 5 skipped, 30 warnings in 1718.32s (0:28:38) ======
```

The relevant change is at least https://github.com/openforcefield/openff-toolkit/pull/1480

--- a/devtools/conda-envs/basic.yaml
+++ b/devtools/conda-envs/basic.yaml
@@ -15,7 +15,6 @@ dependencies:
   - pytest
   - pytest-cov
   - pytest-xdist
-  - codecov
   - requests-mock
 
   - qcengine >=0.25
@@ -28,18 +27,13 @@ dependencies:
 
   # openmmforcefields brings in the full toolkit; if that is ever re-packaged
   # this should be changed to openff-toolkit-base
-  - openff-toolkit >= 0.11.3
-  # argcomplete (brought in via BSE) pulls down totally incorrect builds of `importlib_metadata`,
-  - importlib_metadata >=4
-  - importlib-metadata >=4
-  - rdkit
+  - openff-toolkit >= 0.12
   - pydantic
   - pyyaml
   - qcportal>=0.15.7
   - torsiondrive
   - basis_set_exchange
   - typing-extensions
-  - cachetools
   - h5py>=3.6.0
 
   # Optional

--- a/openff/qcsubmit/utils/utils.py
+++ b/openff/qcsubmit/utils/utils.py
@@ -115,17 +115,19 @@ def get_torsion(bond: off.Bond) -> Tuple[int, int, int, int]:
         If there is more than one possible combination of atoms the heaviest set are selected to be restrained.
     """
 
-    atoms = [bond.atom1, bond.atom2]
-    terminal_atoms = {}
+    atoms: List[off.Atom] = [bond.atom1, bond.atom2]
+    terminal_atoms: Dict[off.Atom, off.atom] = dict()
 
     for atom in atoms:
         for neighbour in atom.bonded_atoms:
             if neighbour not in atoms:
-                if (
-                    neighbour.atomic_number
-                    > terminal_atoms.get(atom, off.Atom(0, 0, False)).atomic_number
-                ):
+                # If we have not seen any possible terminal atoms for this atom, add the neighbour
+                if atom not in terminal_atoms:
                     terminal_atoms[atom] = neighbour
+                # If the neighbour is heavier than the current terminal atom, replace it
+                if neighbour.atomic_number > terminal_atoms.get(atom).atomic_number:
+                    terminal_atoms[atom] = neighbour
+
     # build out the torsion
     return tuple(
         [

--- a/openff/qcsubmit/utils/utils.py
+++ b/openff/qcsubmit/utils/utils.py
@@ -125,7 +125,7 @@ def get_torsion(bond: off.Bond) -> Tuple[int, int, int, int]:
                 if atom not in terminal_atoms:
                     terminal_atoms[atom] = neighbour
                 # If the neighbour is heavier than the current terminal atom, replace it
-                if neighbour.atomic_number > terminal_atoms.get(atom).atomic_number:
+                elif neighbour.atomic_number > terminal_atoms.get(atom).atomic_number:
                     terminal_atoms[atom] = neighbour
 
     # build out the torsion

